### PR TITLE
EXM-14 Create the bar on the right side

### PR DESCRIPTION
**Pull request by thatBeatrice**
- Added the list of recent expenses on the right side of the dashboard.
- Made the list reload and filter expenses based on the top bar's selected tab using `ReloadService`.
- Made the list reload and update the currency based on the dropdown's value.
- Displayed the name of the user on the right side bar from feature **EXM-27**.

### CODE REVIEW

**Comment by thatBeatrice** [comment]
Current problems:
- Dropdown does not have the first currency from the currency list as default value.
- The asynchronous calls are flooding the backend with unnecessary requests (all categories and currencies) every time the component reloads - need to find another place to make certain calls only once.
- (?) As the right sidebar is currently placed in the `AppComponent`, it initializes twice: once when login page is shown and once at home page.

**Review by en-andreea-apriotese — COMMENTED** [review]
Referitor la prima intrebare, nu este necesar sa ii specificam si moneda, daca nu o specificam backend-ul va trimite toate cheltuielile asa cum sunt. Ceea ce este ok, le poti afisa in moneda in care vin, daca nimic nu este selectat.

**Comment by thatBeatrice** [comment]
Current problem: after a certain threshold, the recent expenses list becomes scrollable and expands the whole page as well.

File: expenses-manager-frontend/src/main/frontend/src/app/right-sidebar/right-sidebar.component.html [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  <mat-select *ngIf="router.url == '/home'" [(value)]="selectedCurrencyIndex" (selectionChange)="onCurrencyChange(currencies[$event.value].code)">
+    <mat-option *ngFor="let currency of currencies; let i = index" [value]="i"> 
+      <fa-icon *ngIf="currency.code == 'USD'" [icon]="['fas', 'dollar-sign']"></fa-icon>
+      <fa-icon *ngIf="currency.code == 'EUR'" [icon]="['fas', 'euro-sign']"></fa-icon>
+      {{ currency.code }} 
+    </mat-option>
+  </mat-select>
+</div>
+<mat-divider class="outer-divider"></mat-divider>
+<mat-list *ngIf="router.url == '/home'">
```
Comment (en-andreea-apriotese) [inline expenses-manager-frontend/src/main/frontend/src/app/right-sidebar/right-sidebar.component.html:12]: Aceasta lista ar trebui sa aiba in partea stanga un border, asa cum este in Figma.

File: expenses-manager-frontend/src/main/frontend/src/app/right-sidebar/right-sidebar.component.html [hunk 2]
```diff
@@ -0,0 +1,24 @@
+<div class="horizontal-container"> 
+  <h2>{{ userName }}</h2>
+  <mat-select *ngIf="router.url == '/home'" [(value)]="selectedCurrencyIndex" (selectionChange)="onCurrencyChange(currencies[$event.value].code)">
+    <mat-option *ngFor="let currency of currencies; let i = index" [value]="i"> 
+      <fa-icon *ngIf="currency.code == 'USD'" [icon]="['fas', 'dollar-sign']"></fa-icon>
```
Comment (en-andreea-apriotese) [inline expenses-manager-frontend/src/main/frontend/src/app/right-sidebar/right-sidebar.component.html:5]: As zice ca daca nu avem iconite pentru fiecare moneda, sa le stergem de aici.

File: expenses-manager-frontend/src/main/frontend/src/app/right-sidebar/right-sidebar.component.html [hunk 3]
```diff
@@ -0,0 +1,24 @@
+<div class="horizontal-container">
```
Comment (en-andreea-apriotese) [inline expenses-manager-frontend/src/main/frontend/src/app/right-sidebar/right-sidebar.component.html:1]: De asemenea acest bar ar trebui sa fie pana in capat. Pe ecranele mai mari se termina brusc la un moment dat:

image

Comment (en-andreea-apriotese) [reply expenses-manager-frontend/src/main/frontend/src/app/right-sidebar/right-sidebar.component.html:1]: Poti sa maresti bara cu Day, Week..iar right-sidebar sa ramana la dimensiunea curenta.

**Review by en-andreea-apriotese — APPROVED** [review]

**Pull request merged**